examples/recording_service/leveldb-integration/cpp/LevelDbWriter.cxx

```diff
@@ -0,0 +1,512 @@
+     * times of the plugin. They will represent the start and stop times of the
+     * service closely enough
+     */
+    std::stringstream metadata_filename;
+    metadata_filename << working_dir_ << "/metadata.dat";
+    /* DB options: create the file if not found; rest is fine to keep default */
+    leveldb::Options db_options;
+    db_options.create_if_missing = true;
+    /*
+     * For the best record/replay/convert experience, we do not allow to append
+     * data to an existing database. The following setting will preempt creation
+     * of the database if it already existed. You should choose a different
+     * working directory each time you run the tool
+     */
+    db_options.error_if_exists = true;
+    leveldb::DB *db = nullptr;
+    leveldb::Status db_status =
+            leveldb::DB::Open(db_options, metadata_filename.str(), &db);
+    if (!db_status.ok() || db == nullptr) {
+        std::stringstream log_stream;
+        log_stream << "Failed to open metadata DB file" << std::endl;
+        log_stream << "    File name = " << metadata_filename.str()
+                << std::endl;
+        log_stream << "    LevelDB error = " << db_status.ToString()
+                << std::endl;
+        throw std::runtime_error(log_stream.str());
+    }
+    metadata_db_.reset(db);
+
+    /* Time was returned in seconds. Transform to nanoseconds */
+    current_time *= (int64_t) rti::core::nanosec_per_sec;
+    std::stringstream current_time_stream;
+    current_time_stream << current_time;
+    db_status = metadata_db_->Put(
+            leveldb::WriteOptions(),
+            start_time_key_name(),
+            current_time_stream.str());
+    if (!db_status.ok()) {
+        std::stringstream log_stream;
+        log_stream << "Failed to write start time to metadata DB" << std::endl;
+        log_stream << "    LevelDB error = " << db_status.ToString()
+                << std::endl;
+        throw std::runtime_error(log_stream.str());
+    }
+}
+
+LevelDbWriter::~LevelDbWriter()
+{
+    /* Obtain current time so we can store it as the service's stop time */
+    int64_t current_time = (int64_t) time(nullptr);
+    if (current_time == -1) {
+        std::cerr << "Failed to obtain the current time" << std::endl;
+    }
+    /* Time was returned in seconds. Transform to nanoseconds */
+    current_time *= (int64_t) rti::core::nanosec_per_sec;
+    std::stringstream current_time_stream;
+    current_time_stream << current_time;
+    leveldb::Status db_status = metadata_db_->Put(
+            leveldb::WriteOptions(),
+            end_time_key_name(),
+            current_time_stream.str());
+    if (!db_status.ok()) {
+        std::stringstream log_stream;
+        log_stream << "Failed to write end time to metadata DB" << std::endl;
+        log_stream << "    LevelDB error = " << db_status.ToString()
+                << std::endl;
+        /* Do not throw in a C++ destructor */
+        std::cerr << log_stream.str();
```

Comment:
just as a note: starting next version we'll be able to leverage ROUTING-612.

Reply:
I have just created https://jira.rti.com/browse/RECORD-1163 to track this.